packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -1149,6 +1148,8 @@ export function getChangeHandler(
 	return getFieldKind(fieldKinds, kind).changeHandler;
 }
 
+// TODO: TFieldData could instead just be a numeric ID generated by the CrossFieldTable
+// The CrossFieldTable could have a generic field ID to context table
```

Comment:
I wonder if that might make debugging more awkward because of the indirection. What would be the incentive for using a numeric ID for this?

Reply:
It would mean that each operation (compose, rebase, invert) wouldn't need to pick some arbitrary object to use as the key for the field. I suppose instead of the current pattern of a key and then a table with some data associated with the key, we could use the data objects themselves as the key. That would provide a better debugging experience. The downside to this is that we have to register something to use as a key before running the operation (compose, rebase, invert) while the output of the operation should generally be a required field of the context object.